storage/google/cloud/storage/blob.py

```diff
@@ -1530,25 +1532,15 @@ def update_storage_class(self, new_class, client=None):
         if new_class not in self._STORAGE_CLASSES:
             raise ValueError("Invalid storage class: %s" % (new_class,))
 
-        client = self._require_client(client)
-
-        query_params = {}
-
-        if self.user_project is not None:
-            query_params["userProject"] = self.user_project
-
-        headers = _get_encryption_headers(self._encryption_key)
-        headers.update(_get_encryption_headers(self._encryption_key, source=True))
+        # Update current blob's storage class prior to rewrite
+        self._patch_property('storageClass', new_class)
 
-        api_response = client._connection.api_request(
-            method="POST",
-            path=self.path + "/rewriteTo" + self.path,
-            query_params=query_params,
-            data={"storageClass": new_class},
-            headers=headers,
-            _target_object=self,
-        )
-        self._set_properties(api_response["resource"])
+        # Execute consecutive rewrite operations until operation is done
+        token = None
+        while True:
+            token, _, _ = self.rewrite(self, token=token)
+            if token is None:
+                break
```

Comment:
I think the intent is clearer if we unroll the loop a bit:

```python
        token, _, _ = self.rewrite(self)
        while token is not None:
            token, _, _ = self.rewrite(self, token=token)
```

Reply:
Sure, that makes sense to me.